qiskit/circuit/library/__init__.py

```diff
@@ -94,6 +129,21 @@
 Generalized Gates
 =================
 
+These "gates" (many are :class:`~qiskit.circuit.quantumcircuit.QuantumCircuit` subclasses) allow to
+set the amount of qubits involved at instantiation time.
+
+
+.. jupyter-execute::
+
+    from qiskit.circuit.library import Diagonal
+
+    diagonal = Diagonal([1, 1])
+    print(diagonal.num_qubits)
+
+    diagonal = Diagonal([1, 1, 1, 1])
+    print(diagonal.num_qubits)
```

Comment:
Instead of printing the number of qubits maybe showing the decomposed circuit might be a better example

Reply:
The second one decomposes to two gates and the text points to the amount of qubits. That difference might create some confusing maybe?

```
     ┌────────────┐           
q_0: ┤0           ├───────────
     │  Ucrz(0,0) │┌─────────┐
q_1: ┤1           ├┤ Ucrz(0) ├
     └────────────┘└─────────┘
```

Reply:
OK sure, that works for me. We can always update this again later too if we have a better idea on how to visualize it too